proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+	<description>Registers a service offered by this app on the module</description>
+
+	<param name="appServiceManifest" type="AppServiceManifest" mandatory="false"/>
+
+	
+</function>
+
+
+<function name="PublishAppService" functionID="PublishAppServiceID" messagetype="response">
+	<description>Response to the request to register a service offered by this app on the module</description>
+
+	<param name="success" type="Boolean" platform="documentation" mandatory="true">
+		<description> true if successful; false, if failed </description>
+	</param>
+       
+	<param name="resultCode" type="Result" platform="documentation" mandatory="true">
+		<description>See Result</description>
+		<element name="SUCCESS"/>
+		<element name="INVALID_DATA"/>
+		<element name="OUT_OF_MEMORY"/>
+		<element name="TOO_MANY_PENDING_REQUESTS"/>
+		<element name="APPLICATION_NOT_REGISTERED"/>
+		<element name="GENERIC_ERROR"/>
+	</param>
+	
+	<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+		<description>Provides additional human readable info regarding the result.</description>
+	</param>
+</function>
+
+```
+
+##### Notifying potential consumers - `OnAppServiceRecordUpdated`
+
+Once the service is published to the module, the module will be responsible for ensuring that other connected apps are notified that a new service has registered. This is done through the `OnAppServicePublished` notification. This contains the same manifest that was provided by the original app service provider. 
+
+
+```xml
+<function name="OnAppServiceRecordUpdated" functionID="OnAppServiceRecordUpdatedID" messagetype="notification">
+	<description> This notification includes the data that is updated from the specific service</description>
+	
+	<param name="servicePublished" type="Boolean" mandatory="true"/>
+	<param name="serviceManifest" type="AppServiceManifest" mandatory="true"/>
+	<param name="updateResult" type="Result" mandatory="false"/>
+</function>
+
+```
+
+##### Notifying provider of removal
+
+If for any reason the module determines that an app service provider is no longer to be published and is to be removed it will notify the app service provider through an `OnAppServiceRecordUpdated` with the `servicePublished` boolean set to false. It will be the responsibility of the app service provider to then tear down their service resources on the mobile device side. 
+
+If the service wishes to reregister it can follow the normal flow and pending the `updateResult` param in the notification.
+
+
+#### App Service Consumer
+
+This section will cover how another app can consume the newly published app service.
+
+![App Service Consumer](../assets/proposals/NNNN-app-services/consumer.png "App Service Consumer")
+
+
+##### Get service capabilities
+
+If the app wants to use services published on the device first it must ensure the feature is supported. This requires a modification to the System Capability functionality.
```

Comment:
I think you need a comma in the sentence.  Depending on how it should be phrased, please update to either "...on the device first, it must ensure..." or "...on the device, first it must ensure..."